https://salsa.debian.org/apt-team/python-apt/-/commit/8f9f11b166e54252c0fb545b09c6e388ff4b6825

From 8f9f11b166e54252c0fb545b09c6e388ff4b6825 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <jak@debian.org>
Date: Thu, 21 Dec 2023 15:53:19 +0100
Subject: [PATCH] Actually register apt_pkg.Warning object

Fixes: 81a5896ee410c62ccf140b48142a5358a07331ca
       ("Issue warnings as apt_pkg.Warning instead of discarding when
       there is no error")
---
 python/apt_pkgmodule.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/python/apt_pkgmodule.cc b/python/apt_pkgmodule.cc
index 31f5c5e6..84b1e92a 100644
--- a/python/apt_pkgmodule.cc
+++ b/python/apt_pkgmodule.cc
@@ -915,6 +915,7 @@ extern "C" void initapt_pkg()
    Config->NoDelete = true;
    PyModule_AddObject(Module,"config",Config);
    PyModule_AddObject(Module,"Error",PyAptError);
+   PyModule_AddObject(Module,"Warning",PyAptWarning);
    PyModule_AddObject(Module,"CacheMismatchError", PyAptCacheMismatchError);
 
 
-- 
GitLab
